libs/libcommon/src/libcommon/utils.py

```diff
@@ -49,16 +49,33 @@ def utf8_lead_byte(b: int) -> bool:
     return (b & 0xC0) != 0x80
 
 
-def utf8_byte_truncate(text: str, max_bytes: int) -> str:
-    """If text[max_bytes] is not a lead byte, back up until a lead byte is
-    found and truncate before that character."""
-    utf8 = text.encode("utf8")
-    if len(utf8) <= max_bytes:
-        return text
+class SmallerThanMaxBytesError(Exception):
+    pass
+
+
+def serialize_and_truncate(obj: Any, max_bytes: int) -> str:
+    """
+    Serialize an object as JSON and truncate it if it's bigger than max_bytes
+
+    Args:
+        obj (`Any`): the Python object (can be a primitive type, a list, a dict, etc.)
+        max_bytes (int): the maximum number of bytes
```

Comment:
Not sure if this PR will be approved but then I would add '' to each data type https://github.com/huggingface/datasets-server/pull/2351

Reply:
I think https://github.com/huggingface/datasets-server/pull/2351 will be merged first, I will adapt

Reply:
Done